src/jenkins-enhanced.coffee

```diff
@@ -353,13 +486,86 @@ class HubotJenkinsPlugin extends HubotMessenger
     "HEALTH: #{result}\n"
 
   _getJob: (escape = false) =>
-    job = @msg.match[1].trim()
-
-    # if the provided name is an alias, provide it's mapped job name
-    aliases = @_getSavedAliases()
-    job     = aliases[job] if aliases[job]
-
-    if escape then @_querystring.escape(job) else job
+    # check if its a job we already stored
+    if (typeof(@msg.match[1]) == "object")
+      return @msg.match[1]
+    # check if the user gave us a folder path to follow and a job
+    else
+      # otherwise search for it with what the user gave us
+      # if they gave us a folder use that
+      if (@msg.match[1].indexOf("/") != -1)
+        folderPath = @msg.match[1].split("/")
+        jobName = folderPath[folderPath.length-1]
+        folderPath.splice(folderPath.length-1, 1)
+        if (folderPath.length == 1)
+          folders = []
+          for server in @_serverManager.listServers()
+            folders = folders.concat(server.getFolderByName(folderPath[0]))
+          if (folders.length > 1)
+            @send "There are multiple folders with the name #{folderPath[0]}.  Please use `jenkins list` and an ID instead."
+          else if (folders.length == 1)
+            jobs = folders[0].getJobByName(jobName)
+            if (jobs.length > 1)
+              @send "There are multiple jobs with the name #{jobName} in #{folderPath[0]}.  Please use `jenkins list` and an ID instead."
+            else if (jobs.length == 1)
+              return jobs[0]
+            else 
+              @send "There are no jobs with the name #{jobName} in #{folderPath[0]}."
+          else
+            @send "There are no folders with the name #{folderPath[0]}."
+			
+        else
+          # this is for an absolute path to the folder
```

Comment:
Perhaps take the body of this `else` and wrap into method named `_getJobByAbsolutePath` (or something like that)